### PR TITLE
feat(files): ask for confirm if deleting 5 items or more 

### LINKED ISSUE #32011: File Delete Confirmation Dialog
**Is your feature request related to a problem? Please describe.**
When we right-click on the files we have the last two options from the "Download" and "Delete" list. It is very common for our users to click "Delete" instead of "Download" and the files are deleted directly.

**Describe the solution you'd like**
When we click on "Delete" a popup window will open asking us to confirm if we really want to delete.

**Describe alternatives you've considered**
The "Download" and "Delete" buttons are no longer below each other.

**Additional context**
It's not necessary.

### PR DESCRIPTION
**Pull request by skjnldsv**
Fix https://github.com/nextcloud/server/issues/32011
Needs https://github.com/nextcloud/server/pull/43217

image
@nextcloud/designers

### CODE REVIEW

File: apps/files/src/actions/deleteAction.ts [hunk 1]
```diff
@@ -136,7 +139,34 @@ export const action = new FileAction({
...
+					},
+				)
+				return
+			}
+			resolve(true)
+		})
+
+		// If the user cancels the deletion, we don't want to do anything
+		if (confirm === false) {
+			showError(t('files', 'Cancelled deletion'))
```
Comment (artonge) [inline apps/files/src/actions/deleteAction.ts:166]: No?
```suggestion
			showError(t('files', 'Deletion cancelled'))
```

**Review by artonge — APPROVED** [review]

File: apps/files/src/actions/deleteAction.ts (hunk 1 above)
Comment (skjnldsv) [reply apps/files/src/actions/deleteAction.ts]: Done! :)

**Review by susnux — APPROVED** [review]

**Review by szaimen — APPROVED** [review]
LGTM from design perspective :)

**Comment by AndyXheli** [comment]
Can't wait for this!!!!! Would this also work if a user moves large amounts of data from one share into another ?

**Pull request merged**